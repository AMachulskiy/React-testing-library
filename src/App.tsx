import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import Nav from './components/nav/nav'
import AboutPage from './pages/aboutPage'
import ContactPage from './pages/contactPage'
import MainPage from './pages/mainPage'
import UserPage from './pages/userPage'
import UsersPage from './pages/usersPage'
import store from './store/store'

const App: React.FC = () => {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <Nav />
        <Routes>
          <Route path='/' element={<MainPage />} />
          <Route path='/about' element={<AboutPage />} />
          <Route path='/contact' element={<ContactPage />} />
          <Route path='/users' element={<UsersPage />} />
          <Route path='/users/:id' element={<UserPage />} />
        </Routes>
      </Provider>
    </BrowserRouter>
  )
}

export default App
