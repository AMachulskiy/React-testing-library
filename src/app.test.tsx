import { configure, fireEvent, render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'
import App from './App'

configure({
  asyncUtilTimeout: 2000,
})

describe('Test App', () => {
  beforeEach(() => {
    render(<App />)
  })
  test('Render main page', () => {
    const title = screen.getByTestId(/happy/i)
    expect(title).toBeInTheDocument()
    expect(title).toHaveAttribute('title', 'Happy')
    expect(title).toHaveClass('happy-title fixed')
    expect(title).toHaveStyle({
      color: 'white',
    })
  })
  test('Async findBy', async () => {
    const stateElem = await screen.findByTestId('state-elem')
    expect(stateElem).toBeInTheDocument()
  })
  test('test events', () => {
    const checkbox = screen.getByTestId('checkbox')
    const status = screen.getByTestId('checkbox-status')
    expect(status).toHaveTextContent('UnChecked')
    fireEvent.click(checkbox)
    expect(status).toHaveTextContent('Checked')
    expect(checkbox).toBeInTheDocument()
  })
  test('test snapshot', () => {
    const input = screen.getByTestId('input')
    expect(input).toMatchSnapshot()
  })
  test('test userEvent', async () => {
    const input = screen.getByTestId('input')
    expect(input).toMatchSnapshot()
    fireEvent.input(input, {
      target: {
        value: '138',
      },
    })
    expect(input).toHaveValue('138')
    await userEvent.type(input, 'Boom!')
    expect(screen.getByText(/boom/i)).toHaveTextContent('Boom!')
  })
})
