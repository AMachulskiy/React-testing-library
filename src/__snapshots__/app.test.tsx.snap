// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App test snapshot 1`] = `
<input
  data-testid="input"
  placeholder="Введите ваш email"
  type="text"
  value="test@mail.ru"
/>
`;

exports[`Test App test userEvent 1`] = `
<input
  data-testid="input"
  placeholder="Введите ваш email"
  type="text"
  value="test@mail.ru"
/>
`;

exports[`Test App test userEvent 2`] = `
<input
  data-testid="input"
  placeholder="Введите ваш email"
  type="text"
  value="138"
/>
`;
